File: 0011-container-with-most-water/0011-container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */

var maxArea = function(height) {
    let left = 0
    let right = height.length-1
    let max = 0

    while (left < right) {
        let lowest = Math.min(height[left], height[right])
        let length = right-left
        let current = length*lowest
        // console.log(height[left], height[right], length, lowest, current)
        if (current > max) {
            max = current
        }
        height[left] < height[right] ? left++ : right--
    }
return max
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        
        left = 0
        right = len(height)-1
        max = 0
        
    
        while left < right:
            lowest = min(height[left], height[right])
            length = right-left
            current = length*lowest
            print(height[left], height[right], lowest, current)
            if current > max:
                max = current
            if height[left] < height[right]:
                left +=1
            else:
                right -=1
        return max